Assets/Scripts/DogAI/DogAI.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Author(s): Timothy Ng
+ * Description: Brains behind the dogs actions on the world
+ * Usage: [no notes]
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using k = PPGlobal;
+
+public class DogAI : MonoBehaviour {
+
+    enum State
+    {
+        Idle,
+        Wandering,
+        Eating,
+        Pooping
+    }
+
+    State currentState = State.Wandering;
+    Vector2 wanderCenter;
+    float wanderRadius = 500;
+
+    Vector2 target;
+    int frameCounter = 0;
+
+    int tapCount = 0;
+
+	// Use this for initialization
+	void Start () {
+        wanderCenter = GetComponent<RectTransform>().anchoredPosition;
+        target = GetComponent<RectTransform>().anchoredPosition;
+    }
+	
+	void Update () {
+
+        currentState = DecideState();
+
+        switch (currentState)
+        {
+            case State.Idle:
+                break;
+            case State.Wandering:
+                Wander();
+                break;
+            case State.Eating:
+                break;
+            case State.Pooping:
+                break;
+            default:
+                break;
+        }
+	}
+
+    State DecideState()
+    {
+        if(frameCounter < 120)
+        {
+            frameCounter++;
+            return currentState;
+        }
+        else
+        {
+            frameCounter = 0;
+            switch(Random.Range(0, 2))
```

Comment:
I assume this switch statement is set to two because `Eating` and `Pooping` aren't yet implemented. Instead, can we have an array of `implementedStates` and random select an index in that array?